File: Fibonacci.java
import java.math.BigInteger;
import java.util.ArrayList;
import java.util.List;

public class Fibonacci {
    /* BigInteger keeps the sequence exact for any n, like Python's ints. */
    public static List<BigInteger> fibonacci(int n) {
        List<BigInteger> sequence = new ArrayList<>();
        if (n <= 0) {
            return sequence;
        }
        BigInteger a = BigInteger.ZERO;
        BigInteger b = BigInteger.ONE;
        sequence.add(a);
        for (int i = 1; i < n; i++) {
            BigInteger next = a.add(b);
            a = b;
            b = next;
            sequence.add(a);
        }
        return sequence;
    }

    public static void main(String[] args) {
        int count = 10;
        if (args.length > 0) {
            try {
                count = Integer.parseInt(args[0]);
            } catch (NumberFormatException error) {
                System.out.println("Invalid input '" + args[0] + "', using default of 10.");
            }
        }
        System.out.println("First " + count + " Fibonacci numbers: " + fibonacci(count));
    }
}

File: jay-python/fibonacci.py
"""Program 3: Fibonacci Sequence.

Generates the first N Fibonacci numbers. Default is 10.

    python fibonacci.py 8
"""

import sys


def fibonacci(n: int) -> list:
    """Return the first n numbers of the Fibonacci sequence."""
    if n <= 0:
        return []
    sequence = [0]
    a, b = 0, 1
    for _ in range(n - 1):
        a, b = b, a + b
        sequence.append(a)
    return sequence


def main() -> None:
    count = 10
    if len(sys.argv) > 1:
        try:
            count = int(sys.argv[1])
        except ValueError:
            print(f"Invalid input '{sys.argv[1]}', using default of 10.")
    print(f"First {count} Fibonacci numbers: {fibonacci(count)}")


if __name__ == "__main__":
    main()
